### DISCUSSION TITLE
Reduce size of Infinitime Firmware

### DISCUSSION THREAD

**Discussion by Johnmcenroyy** [InfiniTimeOrg/InfiniTime#1551] [Q&A] [answered]
Hello
Is it possible to disable some apps, for example, Navigation app, for building firmware with debug mode ?

For now it gives an error: region FLASH overflowed with .data and user data
In release mode all is fine, despite using GCC 12.2

Thanks

**Comment by JF002** [comment, marked as answer]
The build system does not provide any way to disable parts of the firmware. That's something we might add in the future, but it's not there yet.

However, you can manually disable apps and fonts by commenting them in the code. That way, the compiler won't add them in the final binary, which will reduce the size of the firmware. For example, the Navigation app contains a lot of icons that can be easily disabled.

**Reply by Johnmcenroyy** [reply]
Hi JF002, thanks, also wanted to add that InfiniTime was succesfully compiled with latest GCC 12.2 and lastest nRF5 SDK 17.1.0

**Reply by FintasticMan** [reply]
While InfiniTime does compile with the latest SDK, last time I checked it doesn't boot. With the latest GCC toolchain, everything does seem to work, but there are some link-time warnings that we are trying to get rid of before updating.

**Reply by Johnmcenroyy** [reply]
So it is more safe to use recommended 
A cross-compiler : ARM-GCC (10.3-2021.10)
The NRF52 SDK 15.3.0 : nRF-SDK v15.3.0

Especially considering that I am trying to boot InfiniTime on Lenovo S2 Smart Watch :)
